spark/src/main/scala/org/apache/comet/serde/aggregates.scala

```diff
@@ -140,7 +140,7 @@ object CometAverage extends CometAggregateExpressionSerde with ShimQueryPlanSerd
       return None
     }
 
-    if (!isLegacyMode(avg)) {
+    if (!avg.evalMode.equals(EvalMode.LEGACY)) {
```

Comment:
No need to use `equals` in Scala.

```suggestion
    if (avg.evalMode != EvalMode.LEGACY) {
```